### ISSUE TITLE
Sub/Component with Kotlin's default (@JvmDefault) interface methods ask to add @Provide for dependency

### ISSUE THREAD

**Issue by kuFEAR** [google/dagger#2313]
On current 2.31.2, I've faced with processor error with warning message about missed `@Provide` for `FeatureComponent`

```kotlin
@UserScope
@Subcomponent(modules = [UserComponentModule::class])
interface UserComponent {
   @JvmDefault fun featureComponent() : FeatureComponent = FeatureComponent(this)
}
```

but it works well for Java

```java
@UserScope
@Subcomponent(modules = {UserComponentModule.class})
public interface UserComponent {
    default OffersComponent featureComponent() { 
         return FeatureComponent(this) 
    }
}
```

**Comment by bcorso** [comment]
Hi @kuFEAR,

It looks like the issue is that the kapt generated java stub for these methods don't have the default modifier (https://youtrack.jetbrains.com/issue/KT-25960), which confuses our annotation processors.

From the bug, it should be fixed in `org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.20`. I've verified that it's fixed in the latest 1.4.21.

**Comment by kuFEAR** [comment]
Thanks! I'll try
